### ISSUE TITLE
Ambilight+Hue not updating light (Hue Bridge) color even though the ui shows the color

### ISSUE THREAD

**Issue by Cludch** [diyhue/diyHue#432]
**Describe the bug**
Using Ambilight+Hue and a Hue Bridge linked with diyHue, the lights stay white (sometimes) and are not updated.
Even though the correct light color (e.g. red when the TV is red, is shown in the diyHue web interface).
My guess is that the update command is being sent to the Hue Bridge or rejected?
Toggling Ambilight+Hue off and on again, makes some lights work. This seems somewhat random as sometimes some lights work and another time none works.

Hue Sync does work great without a huge delay.

**To Reproduce**
Try out Ambilight+Hue with a linked Hue Bridge.

**Expected behavior**
All lights paired with the Hue Bridge, which are selected on the TV, should update the color correctly.

**Logs**
Don't have that to my disposal currently, but it contained no errors and only the PUT (or POST?) request from the TV to update the lights.

**Comment by Sigfrodr** [comment]
Same for me.

**Comment by alexyao2015** [comment]
Not very familiar with that but if it is meant to have the colors continually change, it's possible that there's a rate limit that's being hit. Similarly with hue sync (supported by native and esphome afaik)

**Comment by Cludch** [comment]
I don't think that's the issue as it can work with all lights.
The TV is basically sending the same amount of requests as diyHue does, so I guess there is no rate limiting.

Might just be a simple race condition, when initializing because the lights can work sometimes.

**Comment by alexyao2015** [comment]
The TV sending requests to diyhue is different from diyhue sending requests to the hue bridge.

**Comment by Cludch** [comment]
Sorry, let me rephrase my last reply:

The TV is basically sending the same amount of requests to the Hue Bridge when paired natively as diyHue does, so I guess there is no rate limiting.

**Comment by alexyao2015** [comment]
Personally don't own any hue devices so I can't test but my experience with yeelight tells me there could be a different api used to send requests that isn't rate limited

**Comment by Sigfrodr** [comment]
When controlling the yeelights directly from the app, there is no problem to control the colors of all the lights at the same time.
So APP->HUB->YEELIGHTS is ok, but TV->HUB->YEELIGHTS is not (although the colors effectively show OK in the UI).
I'd guess someting is wrong on the hub side when it handles TV commands.

**Comment by alexyao2015** [comment]
Still not understanding what is going on and what exactly ambilight+hue is. There's no way to reproduce this or understand how to fix it without this basic information.

**Comment by Cludch** [comment]
As stated earlier, I'd be more than happy to provide you with basic information, logs, traces etc.
Please tell me, what kind of information you need.

**Comment by alexyao2015** [comment]
Everything including what this app/device is, how it's setup from a clean setup, bulbs used to test, debug logs, etc

**Comment by Cludch** [comment]
1. What is Ambilight+Hue
A Philips tv feature, which syncs the ambilight to surrounding Hue lights. It seems to work with basic HTTP requests and not the Entertainment API.

2. Setup
The tv is paired with the brige. After that one selects the bulbs that should be used and where they are located (left, right)

3. Used bulbs
Hue E27 4th and 5th gen. Also WS2812B with ESPHome. Philips Hue Iris and Hue LightStrip first gen.

4. Debug logs
Will add them at a later time.

**Comment by coolcoons** [comment]
> Same for me.

and me

**Comment by Sigfrodr** [comment]
not stale

**Comment by Cludch** [comment]
I will provide the logs this weekend.

**Comment by hendriksen-mark** [comment]
Did you restart the diyhue bridge?
i have ambilght+hue now working with a mega2560 with ethernet.
before i used esp8266 via wifi but the esp freeze for a sec now with the mega not.
also often with pairing proces the tv say "check power supply" and the leds stay white and in the app no problems to fix this i have to restart the diyhue bridge and everything works

**Comment by Cludch** [comment]
I did some testing yesterday.
At first I paired the TV with the diyHue bridge and assigned the lights.
However, the lights never change color. Neither the diyHue ones nor the native Hue Bridge devices.

Here are the logs:
`10.0.10.16` is the TV.
[diyHue_logs.txt](https://github.com/diyhue/diyHue/files/5213896/diyHue_logs.txt)


What seems strange is that whenever I re-open the settings for the diyHue bridge on the TV, I have to re-pair it. So it might just loose or reject the connection from the bridge, which might be the reason why the bridge does not receive / process the new color update commands.

**Comment by alexyao2015** [comment]
Try it with the master-refactor branch and esphome lights. What seems to be happening is that the lights did not receive the command to turn on.

**Comment by alexyao2015** [comment]
At this point, there still simply is not enough information to confirm if this is a true bug or user configuration issue. Until that is determined, no work can be done on this.

**Comment by Cludch** [comment]
I will try the master-refactor branch. However, I find it weird that this does not seem to be a bug in your eyes.

The Web UI changes the colors, which are sent from the TV. Which means there are packages sent from the client to the server. This does work with a genuine Hue bridge. But diyHue does not forward the messages to the ESPHome (might work with the branch) or Hue Bridge bulbs.

**Comment by alexyao2015** [comment]
No debugging will be done until:
1. We know that this issue still occurs with the lastest version being master-refactor.
2. We have sufficient logs showing with a single protocol (Best using esphome) that lights change using a standard application such as the Hue app and them not changing with this application.

**Comment by Cludch** [comment]
1. I deployed the new version and will do some testing soon.
2. Will provide logs of that. This does work fine with both ESPHome and Hue Bridge linked devices.

**Comment by Cludch** [comment]
I did some further testing earlier today:

1. Controlling a ESPHome WS2812B strip via the Hue app
[diyhue_light_app.txt](https://github.com/diyhue/diyHue/files/5251703/diyhue_light_app.txt)

2. Controlling a Hue Bridge linked LightStrip via the Hue app (connected to the diyHue bridge)
[hue_bridge_light.txt](https://github.com/diyhue/diyHue/files/5251704/hue_bridge_light.txt)

3. Ambilight+Hue with one ESPHome and one Hue Bridge linked LightStrip.
This worked with the master-refactoring branch. However, the ESPHome light does flash sometime because some transitions are not correct (I guess). But this is mostly fine. The Hue lights are delayed and sometimes do not update (looks like a rate limit?)
However, using Hue Sync on the PC, all Hue linked lights are updating instantly and without any limit or timeout.
[tv_working.txt](https://github.com/diyhue/diyHue/files/5251712/tv_working.txt)

**Comment by AlexKusnezov** [comment]
@alexyao2015 same here I'm on master branch, Ambilight+Hue is set up with diyhue lightstrips, but it's not working when I turn it on in the TV.
Can someone explain whats master-refactor branch and how to switch? There's no easy_install.sh in master-refactor.

**Comment by alexyao2015** [comment]
So what exactly is the deal here? It works fine in the master-refactor branch but didn't in the master branch? 

@AlexKusnezov I suggest reading the getting started documentation again.

**Comment by Cludch** [comment]
Kinda.
The only issue that I faced, is the following:

> However, the ESPHome light does flash sometime because some transitions are not correct (I guess). [...] The Hue lights are delayed and sometimes do not update (looks like a rate limit?)

**Comment by alexyao2015** [comment]
Ok I'm going to say this is effectively a non issue then.

**Comment by alexyao2015** [comment]
I did take a look at those logs after the fact. The logs indicate that they were definitely not from the master-refactor branch. This indicates that this was likely a non issue from that start.

**Comment by Cludch** [comment]
I ported every change from the master-refactor branch to my Docker image.
However, I ignored the new logger.

**Comment by alexyao2015** [comment]
Not sure of the purpose of that, especially since it goes against the docker methodology.

**Comment by GaoMax** [comment]
Hi, I'm having the exact same issue as described above. Running master branch with Ambilight TV and Yeelight Strip + 2 Yeelight Bulbs. Ambilight + Hue only works for one of the Yeelight devices at a time, but the color change can be watched with Hue Essentials App. I tried to install master-refactor, but with the current version light search is not correctly and throws this error with debug mode activated. 
`----------------------------------------
Exception happened during processing of request from ('192.168.188.53', 63736)
Traceback (most recent call last):
  File "/usr/lib/python3.7/socketserver.py", line 650, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python3.7/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python3.7/socketserver.py", line 720, in __init__
    self.handle()
  File "/usr/lib/python3.7/http/server.py", line 426, in handle
    self.handle_one_request()
  File "/usr/lib/python3.7/http/server.py", line 414, in handle_one_request
    method()
  File "/opt/hue-emulator/webServer/server.py", line 551, in do_POST
    Thread(target=lightManager.discover.scan_for_lights).start() #this needs to be fixed for issue #418
AttributeError: module 'lightManager' has no attribute 'discover'
----------------------------------------`

**Comment by alexyao2015** [comment]
Hi! Thanks for letting me know about that bug in master-refactor and providing the appropriate logs. I just pushed out a fix for it, so try using the new version. That issue is unrelated to this one, so in the future, please create a new issue with the logs.

**Comment by GaoMax** [comment]
Hi, thanks for the fix! Search now works on master-refactor, but the issue with Ambilight still remains. I found this exception in debug mode, after the configuration on the tv. During Config all the lights update, but after it this occurs: 
`----------------------------------------
Exception happened during processing of request from ('192.168.188.183', 35526)
Traceback (most recent call last):
  File "/usr/lib/python3.7/socketserver.py", line 650, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python3.7/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python3.7/socketserver.py", line 720, in __init__
    self.handle()
  File "/usr/lib/python3.7/http/server.py", line 428, in handle
    self.handle_one_request()
  File "/usr/lib/python3.7/http/server.py", line 394, in handle_one_request
    self.raw_requestline = self.rfile.readline(65537)
  File "/usr/lib/python3.7/socket.py", line 589, in readinto
    return self._sock.recv_into(b)
ConnectionResetError: [Errno 104] Connection reset by peer
----------------------------------------
`

**Comment by mariusmotea** [comment]
``` Connection reset by peer ``` means there was a timeout with one of the devices from the syncronization process. Probably there is a lazy light that remain behind too much.
